// tex-math-parser/dist/Token.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.typeToOperation = exports.lexemeToType = void 0;
exports.lexemeToType = {
    '=': 2 /* Equals */,
    '+': 3 /* Plus */,
    '-': 4 /* Minus */,
    '*': 5 /* Star */,
    '\\cdot': 5 /* Star */,
    '\\times': 6 /* Times */,
    '^': 8 /* Caret */,
    '/': 7 /* Slash */,
    ',': 9 /* Comma */,
    '{': 10 /* Lbrace */,
    '}': 11 /* Rbrace */,
    '(': 12 /* Lparen */,
    ')': 13 /* Rparen */,
    '|': 14 /* Bar */,
    '&': 15 /* Amp */,
    bmatrix: 34 /* Matrix */,
    '\\\\': 16 /* Dblbackslash */,
    '\\sqrt': 17 /* Sqrt */,
    '\\frac': 18 /* Frac */,
    '\\sin': 19 /* Sin */,
    '\\cos': 20 /* Cos */,
    '\\tan': 21 /* Tan */,
    '\\csc': 22 /* Csc */,
    '\\sec': 23 /* Sec */,
    '\\cot': 24 /* Cot */,
    '\\arcsin': 25 /* Arcsin */,
    '\\arccos': 26 /* Arccos */,
    '\\arctan': 27 /* Arctan */,
    '\\log': 28 /* Log */,
    '\\ln': 29 /* Ln */,
    '\\pi': 30 /* Pi */,
    e: 31 /* E */,
    '\\begin': 32 /* Begin */,
    '\\end': 33 /* End */,
    '\\left': 35 /* Left */,
    '\\right': 36 /* Right */,
    T: 38 /* T */,
    '\\det': 39 /* Det */,
    '\\operatorname': 40 /* Opname */,
    eigenvectors: 42 /* Eigenvectors */,
    eigenvalues: 41 /* Eigenvalues */,
    cross: 43 /* Cross */,
    proj: 44 /* Proj */,
    comp: 45 /* Comp */,
    norm: 46 /* Norm */,
    inv: 47 /* Inv */,
};
/**
 * A mapping from a token type to the operation it represents.
 * The operation is the name of a function in the mathjs namespace,
 * or of a function to be defined in scope (i.e. in the argument to math.evaluate())
 */
exports.typeToOperation = {
    [3 /* Plus */]: 'add',
    [4 /* Minus */]: 'subtract',
    [5 /* Star */]: 'multiply',
    [6 /* Times */]: 'multiply',
    [8 /* Caret */]: 'pow',
    [7 /* Slash */]: 'divide',
    [18 /* Frac */]: 'divide',
    [14 /* Bar */]: 'abs',
    [17 /* Sqrt */]: 'sqrt',
    [19 /* Sin */]: 'sin',
    [20 /* Cos */]: 'cos',
    [21 /* Tan */]: 'tan',
    [22 /* Csc */]: 'csc',
    [23 /* Sec */]: 'sec',
    [24 /* Cot */]: 'cot',
    [25 /* Arcsin */]: 'asin',
    [26 /* Arccos */]: 'acos',
    [27 /* Arctan */]: 'atan',
    [28 /* Log */]: 'log10',
    [29 /* Ln */]: 'log',
    [39 /* Det */]: 'det',
    [42 /* Eigenvectors */]: 'eigenvectors',
    [41 /* Eigenvalues */]: 'eigenvalues',
    [43 /* Cross */]: 'cross',
    [44 /* Proj */]: 'proj',
    [45 /* Comp */]: 'comp',
    [46 /* Norm */]: 'norm',
    [47 /* Inv */]: 'inv',
};
class Token {
    /**
       * A token in a TeX string.
       * @param {string} lexeme string literal of the token
       * @param {TokenType} type type of the token
       * @param {Number} pos position of the token in the input string
       *
       * @constructor Token
       */
    constructor(lexeme, type, pos) {
        this.lexeme = lexeme;
        this.type = type;
        this.pos = pos;
    }
}
exports.default = Token;
